aws/resource_aws_alb_listener_rule.go

```diff
@@ -147,6 +148,15 @@ func resourceAwsAlbListenerRuleRead(d *schema.ResourceData, meta interface{}) er
 	rule := resp.Rules[0]
 
 	d.Set("arn", rule.RuleArn)
+
+	// The listener arn isn't in the response but can be derived from the rule arn:
+	// arn:aws:elasticloadbalancing:us-east-1:012345678912:listener-rule/app/name/0123456789abcdef/abcdef0123456789/456789abcedf1234
+	listenerArn := strings.Replace(*rule.RuleArn, ":listener-rule/", ":listener/", 1)
+	// arn:aws:elasticloadbalancing:us-east-1:012345678912:listener/app/name/0123456789abcdef/abcdef0123456789/456789abcedf1234
+	listenerArn = listenerArn[0:strings.LastIndex(listenerArn, "/")]
+	// arn:aws:elasticloadbalancing:us-east-1:012345678912:listener/app/name/0123456789abcdef/abcdef0123456789
+	d.Set("listener_arn", listenerArn)
```

Comment:
This seems functionally ok, but do you mind decoupling this new logic into a function and adding a test for it?